File: src/normalizeMerchant.ts
export function normalizeMerchant(raw: string): string {
  const text = raw
    .trim()
    .replace(/\s+/g, "")
    .replace(/ｾﾌﾞﾝ[-ー]?ｲﾚﾌﾞﾝ/g, "セブンイレブン")
    .replace(/セブン[-ー]?イレブン/g, "セブンイレブン")
    .replace(/ファミマ/g, "ファミリーマート")
    .replace(/ﾏﾂｷﾖ/g, "マツモトキヨシ")
    .replace(/ドン・キホーテ/g, "ドンキホーテ");

  return text;
}

File: python/kakeibo/normalize_merchant.py
"""Normalize merchant strings (mirror of TS implementation)."""
from __future__ import annotations

import re

_WHITESPACE_RE = re.compile(r"\s+")
_HALF_WIDTH_SEVEN = re.compile(r"ｾﾌﾞﾝ[-ー]?ｲﾚﾌﾞﾝ")
_FULL_WIDTH_SEVEN = re.compile(r"セブン[-ー]?イレブン")


def normalize_merchant(raw: str) -> str:
    text = _WHITESPACE_RE.sub("", raw.strip())
    text = _HALF_WIDTH_SEVEN.sub("セブンイレブン", text)
    text = _FULL_WIDTH_SEVEN.sub("セブンイレブン", text)
    text = text.replace("ファミマ", "ファミリーマート")
    text = text.replace("ﾏﾂｷﾖ", "マツモトキヨシ")
    text = text.replace("ドン・キホーテ", "ドンキホーテ")
    return text
